docarray/array/storage/base/getsetdel.py

```diff
@@ -185,10 +234,15 @@ def _set_doc_attr_by_id(self, _id: str, attr: str, value: Any):
         :param attr: the attribute of document to update
         :param value: the value doc's attr will be updated to
         """
+        if attr == 'id' and value is None:
+            raise ValueError(
+                'pop id from Document stored in a DocumentArray is not allowed'
```

Comment:
This is a weird message, when am I asking to `pop`?

Reply:
fixed